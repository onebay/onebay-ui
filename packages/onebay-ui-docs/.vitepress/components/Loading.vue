<template>
  <div class="loading-container">
    <svg
      class="loading"
      t="1605617717566"
      viewBox="0 0 1024 1024"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
      p-id="1397"
      width="200"
      height="200"
    >
      <path
        d="M910.222 455.111V512H682.667v-56.889h227.555z m-625.778 0V512H56.89v-56.889h227.555z m170.667 227.556H512v227.555h-56.889V682.667z m0-625.778H512v227.555h-56.889V56.89z m341.333 699.733l-39.822 39.822-159.289-159.288 39.823-39.823 159.288 159.29zM369.778 329.956l-39.822 39.822-159.29-159.29 39.823-39.821 159.289 159.289z m-39.822 267.377l39.822 39.823-159.29 159.288-39.821-39.822 159.289-159.289z m426.666-426.666l39.822 39.822-159.288 159.289-39.823-39.822 159.29-159.29z"
        fill="#1296db"
        p-id="1398"
        data-spm-anchor-id="a313x.7781069.0.i0"
        class="selected"
      ></path>
    </svg>
  </div>
</template>
<style lang="scss">
.loading-container {
  position: absolute;
  z-index: 99;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}
.loading {
  width: 3rem;
  height: 3rem;
  animation: myRotate 3s linear infinite;
}
@keyframes myRotate {
  0% {
    -webkit-transform: rotate(0deg);
  }
  50% {
    -webkit-transform: rotate(180deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}
</style>