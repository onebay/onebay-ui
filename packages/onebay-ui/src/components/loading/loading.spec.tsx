import { mount } from '@vue/test-utils'
import Loading from './index'

describe('Loading Snap', () => {
  it('render initial AtLoading', () => {
    const wrapper = mount(Loading)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Loading -- props size', () => {
    const wrapper = mount(<Loading size="15"></Loading>)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Loading -- props color', () => {
    const wrapper = mount(<Loading color="#fff"></Loading>)
    expect(wrapper.element).toMatchSnapshot()
  })
})
