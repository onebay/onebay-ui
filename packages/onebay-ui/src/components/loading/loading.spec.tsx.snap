// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Snap render Loading -- props color 1`] = `
<div
  class="ob-loading"
  style=""
>
  <div
    class="ob-loading__ring"
    style="border: 1px solid #fff; border-color: #fff transparent transparent transparent;"
  />
  <div
    class="ob-loading__ring"
    style="border: 1px solid #fff; border-color: #fff transparent transparent transparent;"
  />
  <div
    class="ob-loading__ring"
    data-id="[object Object]"
    style="border: 1px solid #fff; border-color: #fff transparent transparent transparent;"
  />
</div>
`;

exports[`Loading Snap render Loading -- props size 1`] = `
<div
  class="ob-loading"
  style="width: 15px; height: 15px;"
>
  <div
    class="ob-loading__ring"
    style="width: 15px; height: 15px;"
  />
  <div
    class="ob-loading__ring"
    style="width: 15px; height: 15px;"
  />
  <div
    class="ob-loading__ring"
    data-id="[object Object]"
    style="width: 15px; height: 15px;"
  />
</div>
`;

exports[`Loading Snap render initial AtLoading 1`] = `
<div
  class="ob-loading"
  style=""
>
  <div
    class="ob-loading__ring"
  />
  <div
    class="ob-loading__ring"
    style=""
  />
  <div
    class="ob-loading__ring"
    data-id="[object Object]"
  />
</div>
`;

exports[`Loading Snap render initial AtLoading 2`] = `
<div
  class="ob-loading"
  style=""
>
  <div
    class="ob-loading__ring"
  />
  <div
    class="ob-loading__ring"
    style=""
  />
  <div
    class="ob-loading__ring"
    data-id="[object Object]"
  />
</div>
`;
